External/TestDataSet/rawdata/sub-Sub1/perf/sub-Sub1_asl.json

```diff
@@ -0,0 +1,52 @@
+{
+	"Modality":"MR",
+	"MagneticFieldStrength":3,
+	"ImagingFrequency":127.744,
+	"Manufacturer":"Philips",
+	"ManufacturersModelName":"Ingenuity",
+	"SoftwareVersions":"3.2.2_3.2.2.1",
+	"MRAcquisitionType":"2D",
+	"PulseSequenceDetails":"v3.2.2 Product Sequence",
+	"ScanningSequence":"GR",
+	"SequenceVariant":"SK",
+	"ScanOptions":"FS",
+	"ImageType":["ORIGINAL","PRIMARY","M","FFE","M","FFE"],
+	"SeriesNumber":1901,
+	"AcquisitionTime":[14,18,22,26,30,34,38,42,46],
+	"AcquisitionNumber":19,
+	"AcquisitionVoxelSize":[3.4,3.4,4.5],
+	"SliceThickness":4.5,
+	"SpacingBetweenSlices":4.5,
+	"SAR":0.102452,
+	"EchoTime":0.010487,
+	"RepetitionTime":4.8,
+	"FlipAngle":90,
+	"CoilString":"SENSE-Head-8",
+	"PartialFourier":0.435484,
+	"PercentPhaseFOV":100,
+	"EchoTrainLength":27,
+	"PhaseEncodingSteps":62,
+	"PostLabelingDelay":2.025,
+	"AcquisitionMatrixPE":62,
+	"ReconMatrixPE":64,
+	"PixelBandwidth":3054,
+	"PhaseEncodingAxis":"i",
+	"ImageOrientationPatientDICOM":[0.988042,-0.143706,0.0558729,0.154183,0.918592,-0.363889],
+	"InPlanePhaseEncodingDirectionDICOM":"COL",
+	"ConversionSoftware":"dcm2niix",
+	"ConversionSoftwareVersion":"v1.0.20181125  (JP2:OpenJPEG) (JP-LS:CharLS) GCC5.5.0",
+	"M0Type":"Separate",
+	"readout_dim":"2D",
+	"bReproTesting":0,
+	"VascularCrushing":false,
+	"BackgroundSuppression":true,
+    "BackgroundSuppressionPulseTime":[1710,3142],
+	"Q":{
+		"BackgroundSuppressionNumberPulses":2,
+		"ArterialSpinLabelingType":"CASL",
+		"InitialPostLabelDelay":2.025,
+		"LabelingDuration":1.650,
+		"SliceReadoutTime":0.0437647
+	},
+	"SliceTiming":[0.0437647,0.0437647,0.0437647,0.0437647,0.0437647,0.0437647,0.0437647,0.0437647,0.0437647,0.0437647,0.0437647,0.0437647,0.0437647,0.0437647,0.0437647,0.0437647,0.0437647,0.0437647,0.0437647,0.0437647,0.0437647,0.0437647,0.0437647,0.0437647,0.0437647,0.0437647,0.0437647,0.0437647,0.0437647,0.0437647,0.0437647,0.0437647,0.0437647,0.0437647,0.0437647,0.0437647]
+}
```

Comment:
Missing field "TotalAcquiredPairs" - the same as number of control/label pairs.

Reply:
And after these edits, run
`jsonLocal = xASL_bids_VendorFieldCheck(jsonLocal);`
`            jsonLocal = xASL_bids_JsonCheck(jsonLocal,'ASL');`
and save again

Reply:
60 volumes, so 30 pairs, right? 8c04995

Reply:
30 pairs - that is correct.

Reply:
Still need to run the code above - but of course first loading and then finish by saving - this only works on the Matlab structures, not the files itself - sorry for missing that initially ;) Other than that - the other updates look good.

Reply:
Ran it right now. I got the following message:
```
Missing required ASL fields: RepetitionTimePreparation
Missing Recommended ASL fields: SliceEncodingDirection, LabelingLocationDescription, 
LabelingOrientation, LabelingDistance
The field ArterialSpinLabelingType is PCASL, please check the dependencies below:
The recommended fields are missing: PCASLType, LabelingPulseAverageGradient, 
LabelingPulseMaximumGradient, LabelingPulseAverageB1, LabelingPulseDuration, 
LabelingPulseFlipAngle, LabelingPulseInterval
```

[Here](https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/01-magnetic-resonance-imaging-data.html#_asljson-file) `RepetitionTimePreparation` is described as optional though?

Pushed the current version: 10c4362